flutter_app/lib/widgets/return_to_launch_widget.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:imacs/modules/return_to_launch.dart';
+
+class ReturnToLaunchTemplate extends StatefulWidget {
+  final ReturnToLaunch returnToLaunchCommand;
+  final int systemID;
+  final int componentID;
+
+// Needs the ReturnToLaunch object from the return_to_launch dart
+
+  const ReturnToLaunchTemplate(
+      {Key? key,
+      required this.returnToLaunchCommand,
+      required this.systemID,
+      required this.componentID})
+      : super(key: key);
+
+  @override
+  ReturnToLaunchButton createState() => ReturnToLaunchButton();
+}
+
+class ReturnToLaunchButton extends State<ReturnToLaunchTemplate> {
+  String buttonLabel = "Return to Launch";
+
+  void updateButton() {
+    setState(() {
+      buttonLabel = "Sent"; // Triggers a rebuild
+    });
+  }
+
+  // Returns a button that tells the drone to return to launch
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+        onPressed: () async {
+          bool result = await widget.returnToLaunchCommand
```

Comment:
The reason the text of you button is not changing is that this async code never finished executing. This is actually to be expected since part of the async code is the completer for TCP initialization, which will never complete unless connected to the drone.